captum/optim/_utils/models.py

```diff
@@ -168,3 +172,93 @@ def collect_activations(
     catch_activ = ActivationFetcher(model, targets)
     activ_out = catch_activ(model_input)
     return activ_out
+
+
+class AvgPool2dLayer(torch.nn.Module):
+    """
+    AvgPool2d layer that also zeros padding of a specific value. This
+    layer is meant to be used to replace MaxPool2d layers.
+
+    Args:
+        kernel_size (int or tuple of int): The size of the window to
+            perform average pooling with.
+        stride (int or tuple of int, optional): The stride window size
+            to use.
+        padding (int or tuple of int): The amount of
+            zero padding to add to both sides.
+        ceil_mode (bool, optional): Whether to use ceil or floor for
+            creating the output shape.
+        value (Any): Used to return any padding that's meant to be ignored
+            by pooling layers back to zero.
+    """
+
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, ...]],
+        stride: Optional[Union[int, Tuple[int, ...]]] = None,
+        padding: Union[int, Tuple[int, ...]] = 0,
+        ceil_mode: bool = False,
+        value: Optional[Any] = float("-inf"),
+    ) -> None:
+        super().__init__()
+        self.avgpool = torch.nn.AvgPool2d(
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            ceil_mode=ceil_mode,
+        )
+        self.value = value
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.avgpool(x)
+        if self.value is not None:
+            x[x == self.value] = 0.0
+        return x
+
+
+def max2avg_pool2d(model, value: Optional[Any] = float("-inf")) -> None:
+    """
+    Replace all nonlinear MaxPool2d layers with their linear AvgPool2d equivalents.

```

Comment:
> It looks like here we also do replacement of layers but don't use replace_layers function ?

Yeah I had to make a separate function because the `replace_layers` function doesn't copy the layer parameters.

Reply:
`replace_layers` is a generic name if it doesn't replace any layer then we can probably give a more specific name to it  or actually try to access all parameters and copy all parameters from one module to another.

Reply:
Okay, I think I've figured out to replace layers while attempting to copy over any parameters that are shared!

Reply:
I'll make the change in a future PR, as the current one is focused on weight visualization.

Reply:
@NarineK Let me know what you think of the new `replace_layers` function!